pkg/controller/statusmanager/status_manager.go

```diff
@@ -225,13 +248,29 @@ func (status *StatusManager) SetNotDegraded(statusLevel StatusLevel) {
 func (status *StatusManager) SetDaemonSets(daemonSets []types.NamespacedName) {
 	status.Lock()
 	defer status.Unlock()
-	status.daemonSets = daemonSets
+	out := map[types.NamespacedName]*daemonsetState{}
+	for _, name := range daemonSets {
+		if existing, ok := status.daemonSets[name]; ok {
+			out[name] = existing
+		} else {
+			out[name] = nil
+		}
```

Comment:
That basically boils down to `out[name] = status.daemonSets[name]`

Reply:
hah, so it does.